mapie/quantile_regression.py

```diff
@@ -0,0 +1,323 @@
+            _description_
+        y_pred_up : _type_
+            _description_
+
+        Raises
+        ------
+        Warning
+            If the aggregated predictions of any training sample would be nan.
+        Examples
+        --------
+        >>> import warnings
+        >>> warnings.filterwarnings("error")
+        >>> import numpy as np
+        >>> from mapie.quantile_regression import _check_validity_results
+        >>> y_preds = np.array([[1, 2, 3],[3, 4, 5],[2, 3, 4]])
+        >>> y_pred_low = np.array([4, 3, 2])
+        >>> y_pred_up = np.array([4, 4, 4])
+        >>> try:
+        ...     _check_validity_results(y_preds, y_pred_low, y_pred_up)
+        ... except Exception as exception:
+        ...     print(exception)
+        ...
+        WARNING: Not correct.
+        """
+        if (
+            (y_pred_low >= y_pred_up).all() or
+                (y_preds[2] <= y_pred_low).all() or
+                (y_preds[2] >= y_pred_up).all()):
+            warnings.warn(
+                "WARNING: Not correct."
+            )
+
+    def _check_cv(
+        self,
+        cv: Optional[Union[int, str, BaseCrossValidator]] = None
+    ) -> Union[str, BaseCrossValidator]:
+        if cv is None:
+            cv == "split"
+            return cv
+        if cv == "split":
+            return cv
+        else:
+            raise ValueError(
+                "Invalid cv method."
+            )
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> MapieQuantileRegressor:
+        # Checks
+        self._check_parameters()
+        cv = self._check_cv(self.cv)
+        estimator = self._check_estimator(self.estimator)
+        alpha = self._check_alpha(self.alpha)
+        self._check_size_calib_set(y, self.alpha)
+        alpha_copy = alpha.copy()
+        alpha_copy.append(0.5)
+        # agg_function = self._check_agg_function(self.agg_function)
+        X, y = indexable(X, y)
+        y = _check_y(y)
+        self.n_features_in_ = check_n_features_in(X, cv, estimator)
+        sample_weight, X, y = check_null_weight(sample_weight, X, y)
+        y = cast(NDArray, y)
+        # n_samples = _num_samples(y)
+
+        # Initialization
+        self.estimators_: List[RegressorMixin] = []
+
+        # Work
+        if sample_weight is None:
+            (X_fit, X_calib,
+                y_fit, y_calib) = train_test_split(
+                X,
+                y,
+                test_size=0.5,
+                random_state=42
+            )
+            sample_weight_fit = sample_weight
+        else:
+            (X_fit, X_calib,
+                y_fit, y_calib,
+                sample_weight_fit, sample_weight_calib) = train_test_split(
+                X,
+                y,
+                sample_weight,
+                test_size=0.5,
+                random_state=42
+            )
+        list_estimators = []
+        list_y_preds_calib = []
```

Comment:
Similar to `self.estimators_`, can we use directly `y_preds_calib` defined as a Dictionary ?